src/Units.jl

```diff
@@ -570,8 +570,11 @@ julia> upreferred(A)
 3.1574795718518295e-20 m³·⁰⁵ s⁵·¹ kg⁻³·⁰⁵
 ```
 """
-function nondimensionalize(param::GeoUnit{T,U}, g::GeoUnits{TYPE}) where {T,U,TYPE}
-    if param.isdimensional
+function nondimensionalize(param::GeoUnit{T,U}, g::Union{GeoUnits{TYPE}, Nothing}) where {T,U,TYPE}
+    if isnothing(g)
+        @warn("The input parameter is not being nondimensionalized, as no characteristic units are given")
+        return param_ND = param
+    elseif param.isdimensional
```

Comment:
This is better done using multiple dispatch. You can leave the original method as it is and add a new one

```julia
function nondimensionalize(param::GeoUnit, g::Nothing)
        @warn("The input parameter is not being nondimensionalized, as no characteristic units are given")
        return param
end
```